src/Warpstone/Internal/ParserExpressions/OptionalWhitespacesParser.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using Warpstone.Internal.ParserImplementations;
+
+namespace Warpstone.Internal.ParserExpressions
+{
+    /// <summary>
+    /// Represents a parser that parses (optional) whitespaces.
+    /// </summary>
+    internal sealed class OptionalWhitespacesParser : ParserBase<string>
+    {
+        /// <summary>
+        /// Singleton instance of the parser.
+        /// </summary>
+        public static readonly OptionalWhitespacesParser Instance = new();
```

Comment:
Add private default constructor to prevent other instances being created besides this singleton.